presto-iceberg/src/main/java/com/facebook/presto/iceberg/RegisterTableProcedure.java

```diff
@@ -0,0 +1,176 @@
+                        getFileSystem(clientSession, hdfsEnvironment, schemaTableName, tableLocation),
+                        new Path(metadataDir)));
+
+        metadata.registerTable(clientSession, schemaTableName, metadataLocation);
+    }
+
+    public static FileSystem getFileSystem(ConnectorSession clientSession, HdfsEnvironment hdfsEnvironment, SchemaTableName schemaTableName, String tableLocation)
+    {
+        HdfsContext hdfsContext = new HdfsContext(
+                clientSession,
+                schemaTableName.getSchemaName(),
+                schemaTableName.getTableName(),
+                tableLocation,
+                true);
+
+        try {
+            return ((HadoopExtendedFileSystem) hdfsEnvironment.getFileSystem(hdfsContext, new Path(tableLocation))).getRawFileSystem();
+        }
+        catch (IOException e) {
+            throw new PrestoException(ICEBERG_FILESYSTEM_ERROR, format("Error getting file system at path %s", tableLocation), e);
+        }
+    }
+
+    public static Path resolveLatestMetadataLocation(ConnectorSession clientSession, FileSystem fileSystem, Path metadataPath)
+    {
+        int mostRecentVersion = -1;
+        long lastModifiedTime = -1;
+        Path mostRecentMetadataFile = null;
+        boolean duplicateVersionsExist = false;
+        try {
+            RemoteIterator<LocatedFileStatus> iterator = fileSystem.listFiles(metadataPath, false);
+            while (iterator.hasNext()) {
+                Path fileLocation = iterator.next().getPath();
+                String fileName = fileLocation.getName();
+                if (fileName.endsWith(METADATA_FILE_EXTENSION)) {
+                    int version = parseVersion(fileName);
+                    long modifiedTime = fileSystem.getFileStatus(fileLocation).getModificationTime();
+                    if (version > mostRecentVersion) {
+                        mostRecentVersion = version;
+                        mostRecentMetadataFile = fileLocation;
+                        lastModifiedTime = -1;
+                        duplicateVersionsExist = false;
+                    }
+                    else if (version == mostRecentVersion) {
+                        duplicateVersionsExist = true;
+                        if (modifiedTime > lastModifiedTime) {
+                            lastModifiedTime = modifiedTime;
+                            mostRecentMetadataFile = fileLocation;
+                        }
+                    }
+                }
+            }
+        }
+        catch (IOException e) {
+            throw new PrestoException(ICEBERG_FILESYSTEM_ERROR, format("Unable to find metadata at location %s", metadataPath), e);
+        }
+
+        if (duplicateVersionsExist) {
+            clientSession.getWarningCollector().add(new PrestoWarning(PARSER_WARNING, format("Multiple metadata files of most recent version %d found at location %s. Using most recently modified version", mostRecentVersion, metadataPath)));
```

Comment:
The `PARSER_WARNING` is mostly just a placeholder here - does this warrant creation of a new warning code?

Reply:
Yes--for clarity, let's create a new warning.  Let's say, something like `MULTIPLE_TABLE_METADATA`.